requires.py

```diff
@@ -16,10 +20,11 @@ def check_key(self):
         hookenv.log('Joined munge provider units: {}'.format(joined_units))
         remote_unit = hookenv.remote_unit()
         hookenv.log('Remote unit: {}'.format(remote_unit))
-        remote_munge_key = joined_units[remote_unit].get('munge_key')
+        remote_munge_key = joined_units[remote_unit].received.get('munge_key')
         # update a munge key if it's different from a local one
         # there were no per-application endpoint buckets at the time of writing
         # implemented in Juju
+        self._local_munge_key()
```

Comment:
Might be better to use a property in the if statement:

    @property
    def local_munge_key(self):
        return hookenv.config().get('munge_key')
        
# ...

        if remote_munge_key and remote_munge_key != self.local_munge_key():